### PR TITLE
Update Translate samples to establish canonical template

### PR DESCRIPTION
**Pull request by alixhami**
Follows new code sample rubric. Please review for compliance with the new code sample rubric and express any language-specific considerations that need to be addressed for this to be the canonical sample.

### CODE REVIEW

**Comment by theacodes** [comment]
I really, really do not like the idea of our snippets being embedded in tests. Our tests are complex to run. I pushed back on this explicitly when this change was proposed and asked for user research before implementing this.

I'm am -1 on the structure, but otherwise fine with the contents.

File: translate/cloud-client/snippets_test.py [hunk 1]
```diff
@@ -15,35 +15,109 @@
...
+    # [START translate_text_with_model]
+    import six
+    from google.cloud import translate
+
+    translate_client = translate.Client()
+
+    text = 'Hello world'
+    target = 'is'  # Target must be an ISO 639-1 language code.
+    model = translate.NMT
+    if isinstance(text, six.binary_type):
```
Comment (theacodes) [inline translate/cloud-client/snippets_test.py]: Use a u prefix on text and it'll always be the Unicode type.

File: translate/cloud-client/snippets_test.py [hunk 2]
```diff
@@ -15,35 +15,109 @@
...
 
 def test_translate_text(capsys):
-    snippets.translate_text('is', 'Hello world')
+    # [START translate_text_with_model]
+    import six
+    from google.cloud import translate
+
+    translate_client = translate.Client()
+
+    text = 'Hello world'
```
Comment (beccasaurus) [inline translate/cloud-client/snippets_test.py]: I just reviewed another PR that does the same update to a new canonical template, but the placeholder variables looked different there. There was a `# TODO(developer)` comment. Should that be added above these placeholder variables?

Also, should this be:

```py
text = 'Your text to analyze, e.g. Hello world'
```

File: translate/cloud-client/snippets_test.py [hunk 3]
```diff
@@ -15,35 +15,109 @@
...
+    target = 'is'  # Target must be an ISO 639-1 language code.
+    model = translate.NMT
```
Comment (beccasaurus) [inline translate/cloud-client/snippets_test.py]: What is `NMT`? Can this have a comment describing the placeholder variable?

**Comment by beccasaurus** [comment]
Just want to also add my -1 to the format being proposed

**Comment by engelke** [comment]
Please address comments or close this PR. Thanks.

**Comment by engelke** [comment]
Comments have not been addressed in two months. Reopen if needed.

**Pull request closed without merging**